nservicebus/outbox/index.md

```diff
@@ -79,8 +79,64 @@ In order to gradually convert an entire system from the DTC to the Outbox:
 WARNING: When verifying outbox functionality, it can sometimes be helpful to temporarily [stop the MSDTC Service](https://technet.microsoft.com/en-us/library/cc770732.aspx). This ensures that the Outbox is working as expected, and no other resources are enlisting in distributed transactions.
 
 
+## Message identity and idempotent processing
+
+If handlers are idempotent then outbox might not be required at all. Outbox relies on message identity to guarantee consistency. When using outbox on both the sender and receiver message identifiers are consistent if a handler is invoked more-than-once for a message. 
```

Comment:
I don't like this whole section:

> Outbox relies on message identity to guarantee consistency.

guarantee what consistency? It relies of the message identity to deduplicate.

> When using outbox on both the sender and receiver message identifiers are consistent if a handler is invoked more-than-once for a message. 

I'm wondering about the correctness of this sentence. The outbox should prevent more-than-once invocation of a handler. If a handler is invoked more than once, then only because the first attempt wasn't successful and the potentially sent messages were rolled back. I'm not aware of any mechanism to guarantee deterministic message identifiers.

> If a sender does not use the outbox or when sending a message outside of a handler the sender *must* generate and apply the same message identifier to guarantee consistency at the receiver.

this sentence is also weird. The only reason you'd have to generate deterministic message identifiers is in case there is a risk of sending the same message multiple times. This can only be caused by a rippling effect and and also can only be solved by using determinisitic message identifiers across the whole set of endpoints. I don't think this one sentence here is in any way sufficient documentation on how to address and solve this problem.

> The same message identity is also important to have deduplication working on the infrastructure level on some transport that support this.

too much of a technical detail.

Reply:
> guarantee what consistency? It relies of the message identity to deduplicate.

It relies on the message identity to guard consistency in the datastore, no other values are used for that. Again, outbox doesn't dedupe, more than one copy of the message can exist on the wire, it does nothing to prevent this.


> The outbox should prevent more-than-once invocation of a handler.If a handler is invoked more than once, then only because the first attempt wasn't successful and the potentially sent messages were rolled back. I'm not aware of any mechanism to guarantee deterministic message identifiers.